beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/AbstractBlockPublisher.java

```diff
@@ -59,41 +58,63 @@ public SafeFuture<SendSignedBlockResult> sendSignedBlock(
         .thenCompose(
             signedBlockContainer ->
                 gossipAndImportUnblindedSignedBlock(signedBlockContainer, broadcastValidationLevel))
-        .thenApply(
-            result -> {
-              if (result.isSuccessful()) {
-                LOG.trace(
-                    "Successfully imported proposed block: {}",
-                    maybeBlindedBlockContainer.getSignedBlock().toLogString());
-                dutyMetrics.onBlockPublished(maybeBlindedBlockContainer.getSlot());
-                return SendSignedBlockResult.success(maybeBlindedBlockContainer.getRoot());
-              } else if (result.getFailureReason() == FailureReason.BLOCK_IS_FROM_FUTURE) {
-                LOG.debug(
-                    "Delayed processing proposed block {} because it is from the future",
-                    maybeBlindedBlockContainer.getSignedBlock().toLogString());
-                dutyMetrics.onBlockPublished(maybeBlindedBlockContainer.getSlot());
-                return SendSignedBlockResult.notImported(result.getFailureReason().name());
-              } else {
-                VALIDATOR_LOGGER.proposedBlockImportFailed(
-                    result.getFailureReason().toString(),
-                    maybeBlindedBlockContainer.getSlot(),
-                    maybeBlindedBlockContainer.getRoot(),
-                    result.getFailureCause());
+        .thenCompose(result -> calculateResult(maybeBlindedBlockContainer, result));
+  }
+
+  private SafeFuture<SendSignedBlockResult> calculateResult(
+      final SignedBlockContainer maybeBlindedBlockContainer,
+      final BlockImportAndBroadcastValidationResults blockImportAndBroadcastValidationResults) {
 
-                return SendSignedBlockResult.notImported(result.getFailureReason().name());
+    return blockImportAndBroadcastValidationResults
+        .broadcastValidationResult()
```

Comment:
We have FAILED_BROADCAST_VALIDATION now, what's purpose of this block. Could it be fired earlier? If not, I'd better organize this code more streamlined by checking only blockImportResult

Reply:
if it could fire earlier, I'm good with it, may worth a comment